ecs-init/volumes/ecs_volume_plugin.go

```diff
@@ -275,18 +274,17 @@ func (a *AmazonECSVolumePlugin) Remove(r *volume.RemoveRequest) error {
 
 	// remove the volume information
 	delete(a.volumes, r.Name)
-	seelog.Infof("Saving state after removing volume %s", r.Name)
-	// remove the state of deleted volume
-	err = a.state.removeVolume(r.Name)
-	if err != nil {
-		seelog.Errorf("Error saving state")
-		return err
-	}
 	// cleanup the volume's host mount path
 	err = a.CleanupMountPath(vol.Path)
```

Comment:
should this error be returned? seems like you are ignoring it as well

Reply:
no, this is same as state save. as long as unmount is successful, we should be good

Reply:
We should probably log an error here. If something goes wrong and we can't cleanup: we need to figure out if we need to leave the volume in place or not. 

I'm leaning toward removing the volume, or maybe adding a retry with a force flag depending on how testing works out.  Agent will ensure that dependent processes stopped before destroying the volume.

Reply:
will change warnf to errorf. also this just removes the volume mount path. even if its not removed, a new volume with the same volume name can still be mounted to the same path. 

with respect to adding retries for remove/unmount volume failure, i prefer not to do that and follow what docker does , i.e failing it. there could be various reasons such as some other container is consuming the volume/mount helper not running etc, which plugin cannot handle.